shared/chat/conversation/messages/index.js

```diff
@@ -1,80 +1,151 @@
-    case 'header':
-      return <Header messageKey={messageKey} />
-    case 'outboxIDAttachment': // fallthrough
-    case 'messageIDAttachment':
-      return (
-        <Wrapper
-          innerClass={Attachment}
-          isSelected={isSelected}
-          measure={measure}
-          messageKey={messageKey}
-          onAction={onAction}
-          onShowEditor={onShowEditor}
-          prevMessageKey={prevMessageKey}
-        />
-      )
-    case 'error': // fallthrough
-    case 'errorInvisible': // fallthrough
-    case 'messageIDError':
-      return <ErrorMessage messageKey={messageKey} />
-    case 'outboxIDText': // fallthrough
-    case 'messageIDText':
-      return (
-        <Wrapper
-          innerClass={TextMessage}
-          isSelected={isSelected}
-          measure={measure}
-          messageKey={messageKey}
-          onAction={onAction}
-          onShowEditor={onShowEditor}
-          prevMessageKey={prevMessageKey}
-        />
-      )
-    case 'supersedes':
-      return <ProfileResetNotice />
-    case 'timestamp':
-      return <Timestamp messageKey={messageKey} />
-    case 'messageIDUnhandled':
+  showResetParticipants: ?Types.ConversationIDKey,
+  showSuperseded: ?Types.ConversationIDKey,
+}
+
+class MessageFactory extends React.PureComponent<Props> {
+  render() {
+    if (this.props.showResetParticipants) {
+      return <ResetUser conversationIDKey={this.props.showResetParticipants} />
+    }
+    if (this.props.showSuperseded) {
+      return <OldProfileReset conversationIDKey={this.props.showSuperseded} />
+    }
+    if (!this.props.message) {
       return null
+    }
+
+    switch (this.props.message.type) {
+      case 'text':
+        return (
+          <Wrapper
+            innerClass={TextMessage}
+            isEditing={this.props.isEditing}
+            isSelected={this.props.isSelected}
+            message={this.props.message}
+            previous={this.props.previous}
+          />
+        )
+      case 'attachment':
+        return (
+          <Wrapper
+            innerClass={Attachment}
+            isEditing={this.props.isEditing}
+            isSelected={this.props.isSelected}
+            message={this.props.message}
+            previous={this.props.previous}
+          />
+        )
+      case 'systemInviteAccepted':
+        return <SystemInviteAccepted message={this.props.message} />
+      case 'systemSimpleToComplex':
+        return <SystemSimpleToComplex message={this.props.message} />
+      case 'systemGitPush':
+        return <SystemGitPush message={this.props.message} />
+      case 'systemAddedToTeam':
+        return <SystemAddedToTeam message={this.props.message} />
+      case 'systemJoined':
+        return <SystemJoined message={this.props.message} />
+      case 'systemText':
+        return <SystemText message={this.props.message} />
+      case 'systemLeft':
+        return <SystemLeft message={this.props.message} />
+      // case 'error':
```

Comment:
What's up with this?

Reply:
this is fixed in part 10. errors are just treated as type Text w/ an error Reason now. there's no specific error type anymore